### DISCUSSION TITLE
Seeking Feedback / Insights on Apache AGE

### DISCUSSION THREAD

**Discussion by aked21** [apache/age#1704] [General]
### **We're seeking your feedback on Apache AGE (A Graph Extension for PostgreSQL) to enhance its features and usability.** 

If you're a developer, database enthusiast, or exploring graph databases, your input is vital for our improvement. 

**Please share any thoughts on enhancing Apache AGE.** 

A useful feedback submission should include a clear description of the issue or suggestion, steps to reproduce the issue, expected vs. actual behavior, environment details, the issue or suggestion's impact, and, if possible, relevant screenshots or logs. 

**Your insights help us identify what needs improvement and prioritize future updates.**

**Comment by JYage** [comment]
Hello!

Thank you so much for opening the door for people to share feedback on Apache AGE!

I am personally very passionate about learning more about GDB and have come across Apache AGE.
I would like to share my feedback on the AGE manual.
It already looks great! At the same time, it appears to be very technically written, which could possibly make it difficult to read and understand for people who may not be experts in the field. So I think that it would be even better if it could be written more easily so that more people, especially, those who are not from a technical background, would be able to understand Apache AGE more easily!

Thank you so much!

**Comment by M-Bostone** [comment]
hello, 
when are you going to launch AGE for windows or it was never meant for windows

**Reply by markgomer** [reply]
There's a windows version installer in this site:
https://agedb.io/community.jsp

**Comment by M-Bostone** [comment]
thanks marco

**Comment by AneeqaCodes** [comment]
Improve error handling and provides more descriptive error messages would greatly enhance the user experience. Clear error messages help users troubleshoot issues effectively, reducing frustration and improving overall usability.
Simplifying the installation process through pre-built binaries or Docker images would lower the barrier to entry for new users and encourage wider adoption of Apache AGE.

**Reply by markgomer** [reply]
Hello @AneeqaCodes, I agree with your suggestions. By the way, do you know there's already a docker image of Apache AGE? Take a look here: https://hub.docker.com/r/apache/age

**Reply by AneeqaCodes** [reply]
@markgomer Thank you for highlighting the availability of the Apache AGE Docker image. It's an invaluable resource for simplifying the deployment process and ensuring accessibility across various environments.

**Reply by mingfang** [reply]
I have a Dockerfile (https://github.com/mingfang/docker-postgres/blob/master/Dockerfile) to build a Postgres image that contains a set of extensions that I use, including AGE.

**Reply by AneeqaCodes** [reply]
@mingfang I will definitely check out your Dockerfile. Thanks for sharing.

**Comment by mingfang** [comment]
Please please replace `agtype` with `jsonb`.
The biggest advantage with AGE over Neo4j is its native integration with Postgres.
But the use of `agtype` makes it almost impossible to use with many other Postgres extensions.
One such use case is to do bm25 search with ParadeDB https://docs.paradedb.com/search/quickstart.
ParadDB can index and search over `jsonb` fields but not `agtype`.

**Reply by markgomer** [reply]
@jrgemignani what do you think?

**Reply by AneeqaCodes** [reply]
@jrgemignani yes you are right JSONB offers broader compatibility and support within the Postgres ecosystem, making it easier to integrate with other extensions and tools also.

**Comment by salahuddibitnine** [comment]
ApacheAge is an impressive product, especially given the growing demand for graph databases. I believe ApacheAge has the potential to make a significant impact in the market. After installing and testing it, I noticed that its performance on browsers could be enhanced. I would be delighted to contribute to its database performance improvements. I came across a GitHub project focused on Age Performance improvement but am unsure if it is publicly accessible.
